src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -244,16 +245,34 @@ ReplicateShardToAllWorkers(ShardInterval *shardInterval)
 
 		if (targetPlacement == NULL || targetPlacement->shardState != FILE_FINALIZED)
 		{
+			uint64 placementId = 0;
+
 			SendCommandListToWorkerInSingleTransaction(nodeName, nodePort, tableOwner,
 													   ddlCommandList);
 			if (targetPlacement == NULL)
 			{
-				InsertShardPlacementRow(shardId, INVALID_PLACEMENT_ID, FILE_FINALIZED, 0,
+				placementId = GetNextPlacementId();
+				InsertShardPlacementRow(shardId, placementId, FILE_FINALIZED, 0,
 										nodeName, nodePort);
 			}
 			else
 			{
-				UpdateShardPlacementState(targetPlacement->placementId, FILE_FINALIZED);
+				placementId = targetPlacement->placementId;
+				UpdateShardPlacementState(placementId, FILE_FINALIZED);
```

Comment:
While looking into to this PR, I realized that the following if statement is more extensive that what we need:
```C
if (targetPlacement == NULL || targetPlacement->shardState != FILE_FINALIZED)
```

There is no case that ` targetPlacement->shardState != FILE_FINALIZED` can be true. Reference table placements should always have ` targetPlacement->shardState == FILE_FINALIZED`.

This is relevant to this PR since then we would never need upsert placement code. So, could you add another commit (first commit of this PR) such that it removes `targetPlacement->shardState != FILE_FINALIZED` from the if statement. Then, update the code as following (or whatever style you wish):

```C
if (targetPlacement == NULL)
{
...

    if (shouldSync)
    {
         // send insert command to the WORKERS_WITH_METADATA
    }
}
else
{
    Assert (shardState == FILE_FINALIZED);
}

```

I'm requesting this change because as this if statement confused you, extending the unnecessary parts could continue to confuse the future readers.